### ISSUE TITLE
PHP: provide option for user to choose PHP implementation vs C extension of Protobuf

### ISSUE THREAD

**Issue by stanley-cheung** [grpc/grpc#11112]
Currently, the gRPC composer package `grpc/grpc` requires the `google/protobuf`composer package, which is a pure PHP implementation of protocol buffers.

We have observed drastic performance improvement by switching to the [C extension](https://pecl.php.net/package/protobuf) implementation of protobuf.

Let's drop the
```
  "require": {
    "google/protobuf": "^v3.3.0"
  },
```
bit from our grpc `composer.json` file. 

After that, we need to update the documentation to show users how to pick and choose which protobuf implementation to use

To use the pure PHP implementation, add this to the user's project `composer.json` file:
```
  "require": {
    "grpc/grpc": "^v1.3.0",
    "google/protobuf": "^v3.3.0"
  },
```
basically making choosing the PHP implementation of protobuf an explicit decision rather than implicit.

To use the protobuf C extension, the user can have this in their `composer.json` file:
```
  "require": {
    "grpc/grpc": "^v1.3.0"
  },
```

Now instead of installing just the gRPC C extension, install the protobuf C extension as well

```
$ sudo pecl install grpc
$ sudo pecl install protobuf
```

The application code of invoking gRPC and protobuf API calls are completely transparent for both protobuf implementations.

**Comment by bshaffer** [comment]
This is great, but we should consider add the following to the `composer.json`:

```json
"suggest": {
    "ext-protobuf": "For better performance, install the protobuf C extension",
    "google/protobuf": "To get started using groc quickly, install the native protobuf library"
}
```

**Comment by bshaffer** [comment]
If these classes are loaded in the C extension, then the composer autoloader should not even kick in, so it should be OK to require the protobuf library AND have the C extension enabled. The only reason this would not be the case is if the autoloading has been implemented improperly.

**Comment by bshaffer** [comment]
see https://github.com/google/protobuf/issues/3107
